aptos-move/move-examples/dao/ft_dao/sources/ft_dao.move

```diff
@@ -0,0 +1,580 @@
+        total_yes: u64,
+        /// Total no notes
+        total_no: u64,
+        /// locked token during voting
+        votes: Table<address, Token>,
+    }
+
+    struct RegisteredDAOs has key {
+        /// Key is the address of DAO creator's account and value is the signer capability of its resource account
+        /// Only one DAO per admin address for now.
+        accounts_signer_caps: Table<address, SignerCapability>,
+        res_acct_to_src_acct: Table<address, address>,
+    }
+
+
+    /// Initialize the DAO platform
+    fun init_module(account :&signer) {
+        move_to(
+            account,
+            RegisteredDAOs {
+                accounts_signer_caps: table::new(),
+                res_acct_to_src_acct: table::new(),
+            }
+        );
+    }
+
+    /// Creator creates a DAO on the platform
+    public entry fun create_dao(
+        admin: &signer, // dao creator
+        name: String, // name of the DAO
+        threshold: u64, // the absolute number of token required to resolve a proposal
+        voting_duration: u64,
+        creator: address, // this is the creator address of goverance token
+        collection_name: String, // collection of the fungible token
+        token_name: String, // token name
+        min_required_proposer_voting_power: u64, // minimal token required to vote
+    ) acquires RegisteredDAOs {
+
+        // create a resource account
+        let seed = bcs::to_bytes(&name);
+        vector::append(&mut seed, bcs::to_bytes(&creator));
+        vector::append(&mut seed, bcs::to_bytes(&collection_name));
+
+        let (res_signer, res_cap) = account::create_resource_account(admin, seed);
+        let res_addr = signer::address_of(&res_signer);
+        let src_addr = signer::address_of(admin);
+        join_platform(src_addr, res_addr, res_cap);
+
+        assert!(string::length(&name) < 128, error::invalid_argument(ESTRING_TOO_LONG));
+
+        move_to(
+            &res_signer,
+            DAO {
+                name,
+                resolve_threshold: threshold,
+                governance_token: GovernanceToken { creator, collection: collection_name, name: token_name },
+                voting_duration,
+                min_required_proposer_voting_power,
+                next_proposal_id: 0,
+            },
+        );
+        move_to(
+            &res_signer,
+            Proposals {
+                proposals: table::new()
+            }
+        );
+        move_to(
+          &res_signer,
+            ProposalVotingStatistics {
+                proposals: table::new()
+            }
+        );
+
+        emit_create_dao_event(
+            &res_signer,
+            name,
+            threshold,
+            voting_duration,
+            min_required_proposer_voting_power,
+            creator,
+            collection_name,
+            token_name,
+        )
+    }
+
+    /// Only DAO Goverance token holders can create proposal
+    public entry fun create_proposal(
```

Comment:
I wonder if it's a problem that we don't lock the coins belonging to the proposer. This can lead to a misalignment of incentives between proposer and the DAO and enable edge case behaviors such as malicious users trying to spam the system with proposals. What are your thoughts here?